reduction_scripts/reduce_data.py

```diff
@@ -0,0 +1,848 @@
+#! /usr/bin/env python
+
+import sys
+import os
+import pickle
+from astropy.io import fits as pyfits
+import pywifes
+import gc
+import datetime
+import numpy as np
+from data_classifier import classify
+import json
+
+#------------------------------------------------------------------------
+start_time = datetime.datetime.now()
+#------------------------------------------------------------------------
+
+# Read the raw data directory from command line
+data_dir = os.path.abspath(sys.argv[1])+'/'
+
+
+# Classify all raw data (red and blue arm)
+obs_metadatas = classify(data_dir)
+
+
+for arm in obs_metadatas.keys():
+
+    obs_metadata = obs_metadatas[arm]
+    # Get the project directory from the file location 
+    proj_dir = os.path.dirname(__file__)
+    
+    # Check observing mode
+    sci_im = obs_metadatas[arm]['sci'][0]['sci'][0]+'.fits'
+    if pywifes.is_nodshuffle(data_dir+sci_im):
+        obs_mode = 'ns'
```

Comment:
Just seen that `ns` is used around the place, so it's reasonably standard. Up to you.

Reply:
Yes, `ns` is everywhere, even in the `.json` files :), that's why we just stuck to it. Let's leave it as is for now. If the users report any issue because of that, we can change it later on. About `obs`, this is pretty standard in astronomy so don't think any user will have problems with that.

Reply:
Yup, that sounds right to me. 

The test is more or less "Did I have any trouble expanding the abbreviation in my head?". If it took you a moment, or if you find it hard to skim read, the abbreviation probably isn't worth it :p

Same test if names are overly verbose. If you can't take in a variable name at a glance, maybe it can probably be more concise.

'ns' got me for a moment because I was thinking 'neutron star', but it's not a big deal.